### PR TITLE
add the logic of estimate the max transactions can be sealed to a block

### PR DESCRIPTION
**Pull request by cyjseagull**
add the logic of estimate the max transactions can be sealed to a block

### CODE REVIEW

File: libconsensus/pbft/PBFTSealer.cpp [hunk 1]
```diff
@@ -97,5 +97,99 @@ void PBFTSealer::stop()
...
+    /// if the current maxBlockCanSeal is larger than m_lastTimeoutTx, return directly
+    if (m_lastTimeoutTx != 0 && maxBlockCanSeal() >= m_lastTimeoutTx)
+    {
+        return;
+    }
+    /// increase the maxBlockCanSeal when its current value is smaller than the
+    /// last-timeout-tx-number
+    {
+        WriteGuard l(x_maxBlockCanSeal);
+        m_maxBlockCanSeal += (0.5 * m_maxBlockCanSeal);
```
Comment (wheatli) [inline libconsensus/pbft/PBFTSealer.cpp]: make the 0.5 can be change in config.ini

File: libconsensus/Sealer.h [hunk 2]
```diff
@@ -195,6 +204,9 @@ class Sealer : public Worker, public std::enable_shared_from_this<Sealer>
     /// handler
     Handler<> m_tqReady;
     Handler<int64_t> m_blockSubmitted;
+
+    uint64_t m_maxBlockCanSeal;
```
Comment (wheatli) [inline libconsensus/Sealer.h]: plz give some comment for this member

File: libconsensus/Sealer.h [hunk 3]
```diff
@@ -114,7 +114,11 @@ class Sealer : public Worker, public std::enable_shared_from_this<Sealer>
     virtual bool shouldWait(bool const& wait) const;
     /// load transactions from transaction pool
     void loadTransactions(uint64_t const& transToFetch);
-    virtual uint64_t calculateMaxPackTxNum() { return m_consensusEngine->maxBlockTransactions(); }
+    virtual void calculateMaxPackTxNum(uint64_t& maxBlockCanSeal)
```
Comment (wheatli) [inline libconsensus/Sealer.h]: where does it be called ?

File: libconsensus/pbft/PBFTSealer.cpp (hunk 1 above)
Comment (cyjseagull) [reply libconsensus/pbft/PBFTSealer.cpp]: already add `enable_dynamic_block_size` and `block_size_increase_ratio` to control the increase speed, please refer to https://github.com/FISCO-BCOS/FISCO-BCOS/pull/695/files#diff-708d2961ab288bc22804247af5e4c497R192

File: libconsensus/Sealer.h (hunk 2 above)
Comment (cyjseagull) [reply libconsensus/Sealer.h]: done, please refer to https://github.com/FISCO-BCOS/FISCO-BCOS/pull/695/files#diff-7169384e87cdf92ca9a6f534a71de137R206

File: libconsensus/Sealer.h (hunk 3 above)
Comment (cyjseagull) [reply libconsensus/Sealer.h]: removed.

**Pull request merged**